src/QueryFilterHelpers.ts

```diff
@@ -0,0 +1,604 @@
+}
+
+/**
+ * Parses an ELM equivalent expression into a filter. This currently handles checks against literals and checks against
+ * a concept reference.
+ *
+ * @param equal The ELM equivalent clause to parse.
+ * @param library The library the clause resides in.
+ * @returns The filter representation.
+ */
+function interpretEquivalent(equal: ELMEquivalent, library: ELM): EqualsFilter | InFilter | UnknownFilter {
+  let propRef: ELMProperty | null = null;
+  if (equal.operand[0].type == 'FunctionRef') {
+    propRef = interpretFunctionRef(equal.operand[0] as ELMFunctionRef);
+  } else if (equal.operand[0].type == 'Property') {
+    propRef = equal.operand[0] as ELMProperty;
+  }
+
+  if (propRef == null) {
+    console.warn('could not resolve property ref for Equivalent');
+    return { type: 'unknown' };
+  }
+
+  if (equal.operand[1].type == 'Literal') {
+    const literal = equal.operand[1] as ELMLiteral;
+    if (propRef.scope && literal.value) {
+      return {
+        type: 'equals',
+        alias: propRef.scope,
+        value: literal.value,
+        attribute: propRef.path,
+        localId: equal.localId
+      };
+    } else {
+      console.warn('Property reference scope or literal value were not found.');
+      return { type: 'unknown' };
+    }
+  }
+
+  if (equal.operand[1].type == 'ConceptRef') {
+    const conceptRef = equal.operand[1] as ELMConceptRef;
+    if (propRef.scope) {
+      return {
+        type: 'in',
+        alias: propRef.scope,
+        attribute: propRef.path,
+        valueCodingList: getCodesInConcept(conceptRef.name, library),
+        localId: equal.localId
+      };
+    } else {
+      console.warn('Property reference scope was not found.');
+      return { type: 'unknown' };
+    }
+  }
+  return { type: 'unknown' };
+}
+
+/**
+ * Gets a list of codes in a CQL concept reference by name. These are returned as
+ * FHIR Codings.
+ *
+ * @param name Name of the concept.
+ * @param library The library elm the concept should be found in.
+ * @returns A list of codings in the concept
+ */
+function getCodesInConcept(name: string, library: ELM): R4.ICoding[] {
+  const concept = library.library.concepts?.def.find(concept => concept.name === name);
+  if (concept) {
+    const codes: R4.ICoding[] = [];
+    concept.code.map(codeRef => {
+      const code = library.library.codes?.def.find(code => code.name == codeRef.name);
+      if (code) {
+        codes.push({
+          code: code?.id,
+          system: library.library.codeSystems.def.find((systemRef: any) => code?.codeSystem.name === systemRef.name).id
+        });
+      }
+    });
+    return codes;
+  }
+  return [];
+}
+
+/**
+ * Parses an ELM equivalent expression into a filter. This currently only handles checks against literal values.
+ *
+ * @param equal The equal expression to be parsed.
+ * @returns Filter representing the equal filter.
+ */
+function interpretEqual(equal: ELMEqual): EqualsFilter | UnknownFilter {
```

Comment:
Forgive me for asking, but do we care about being able to support equality for concepts? I know it's not convention to use this typically. I'm not sure if we've ever seen it in a measure before or not either but idk

Reply:
I'm not quite sure what the question here is. This implementation only checks for equality to a literal. Comparisons to concepts or codes should always be done with equivalence.

Reply:
I'm basically asking if the measures will only and always use equivalence for codes, which it sounds like they will, so we're good.